contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1776,7 +1778,8 @@ tsql_select_common_type_hook(ParseState *pstate, List *exprs, const char *contex
 			strncmp(context, "INTERSECT", strlen("INTERSECT")) == 0 ||
 			strncmp(context, "EXCEPT", strlen("EXCEPT")) == 0 ||
 			strncmp(context, "VALUES", strlen("VALUES")) == 0 ||
-			strncmp(context, "UNION/INTERSECT/EXCEPT", strlen("UNION/INTERSECT/EXCEPT")) == 0)
+			strncmp(context, "UNION/INTERSECT/EXCEPT", strlen("UNION/INTERSECT/EXCEPT")) == 0 ||
+			strncmp(context, "CASE", strlen("CASE")) == 0)
```

Comment:
Suggestion: can we rewrite this [like](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/e57f03a5585b21043b6b26ab23894068e9250fbb/src/backend/nodes/readfuncs.c#L361)
Get length of context once and then use it to filter out most conditions. Would save some time.

Reply:
@Yvinayak07 Please address this.

Reply:
Fixed this change